flink-java/src/main/java/org/apache/flink/api/java/typeutils/PojoField.java

```diff
@@ -68,4 +72,24 @@ private void readObject(ObjectInputStream in)
 	public String toString() {
 		return "PojoField " + field.getDeclaringClass() + "." + field.getName() + " (" + type + ")";
 	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (obj instanceof PojoField) {
```

Comment:
You're right. Will refactor the public access to the fields and include the field in the `equals` and `hashCode` calculation.